plugins/file_chooser/gtk/file_chooser_plugin.cc

```
@@ -0,0 +1,198 @@
+
+  g_autoptr(GtkFileChooserNative) dialog =
+      GTK_FILE_CHOOSER_NATIVE(gtk_file_chooser_native_new(
+          title, nullptr, action, confirm_button_text, "_Cancel"));
+  gtk_window_set_transient_for(GTK_WINDOW(dialog), self->window);
+
+  value = fl_value_lookup_string(properties, kAllowsMultipleSelectionKey);
+  if (fl_value_get_type(value) == FL_VALUE_TYPE_BOOL) {
+    gtk_file_chooser_set_select_multiple(GTK_FILE_CHOOSER(dialog),
+                                         fl_value_get_bool(value));
+  }
+
+  value = fl_value_lookup_string(properties, kCanChooseDirectoriesKey);
+  if (fl_value_get_type(value) == FL_VALUE_TYPE_BOOL &&
+      fl_value_get_bool(value)) {
+    gtk_file_chooser_set_action(GTK_FILE_CHOOSER(dialog),
+                                GTK_FILE_CHOOSER_ACTION_SELECT_FOLDER);
+  }
+
+  value = fl_value_lookup_string(properties, kInitialDirectoryKey);
+  if (fl_value_get_type(value) == FL_VALUE_TYPE_STRING) {
+    gtk_file_chooser_set_current_folder(GTK_FILE_CHOOSER(dialog),
+                                        fl_value_get_string(value));
+  }
+
+  value = fl_value_lookup_string(properties, kInitialFileNameKey);
+  if (fl_value_get_type(value) == FL_VALUE_TYPE_STRING) {
+    gtk_file_chooser_set_current_name(GTK_FILE_CHOOSER(dialog),
+                                      fl_value_get_string(value));
+  }
+
+  value = fl_value_lookup_string(properties, kAllowedFileTypesKey);
+  if (fl_value_get_type(value) == FL_VALUE_TYPE_LIST) {
+    for (size_t i = 0; i < fl_value_get_length(value); i++) {
+      FlValue* filter_info = fl_value_get_list_value(value, i);
+
+      if (fl_value_get_type(filter_info) != FL_VALUE_TYPE_LIST ||
+          fl_value_get_length(filter_info) != 2 ||
+          fl_value_get_type(fl_value_get_list_value(filter_info, 0)) !=
+              FL_VALUE_TYPE_STRING ||
+          fl_value_get_type(fl_value_get_list_value(filter_info, 1)) !=
+              FL_VALUE_TYPE_LIST)
+        continue;
```

We should make bad data an error instead of silently dropping it.